api/v1alpha1/hostedcluster_types.go

```diff
@@ -47,6 +48,22 @@ type ClusterNetworking struct {
 	MachineCIDR string `json:"machineCIDR"`
 }
 
+// ProviderCredentialsSpec describes cloud provider credentials used in various
+// contexts by HyperShift.
+type ProviderCredentialsSpec struct {
+	// KubeCloudControllerCreds is a reference to a secret containing cloud
+	// credentials with permissions matching the Kube cloud controller policy.
+	// The secret should have exactly one key, `credentials`, whose value is
+	// the platform specific credentials content.
+	KubeCloudControllerCreds corev1.LocalObjectReference `json:"kubeCloudControllerCreds"`
+
+	// KubeCloudControllerCreds is a reference to a secret containing cloud
+	// credentials with permissions matching the noe pool management policy.
+	// The secret should have exactly one key, `credentials`, whose value is
+	// the platform specific credentials content.
+	NodePoolManagementCreds corev1.LocalObjectReference `json:"nodePoolManagementCreds"`
```

Comment:
Also probably required?

Reply:
s/KubeControllerCreds/NodePoolManagementCreds/ in godoc